src/Digdir.Domain.Dialogporten.Application/Features/V1/EndUser/Dialogs/Queries/Search/SearchDialogQuery.cs

```diff
@@ -191,7 +191,10 @@ public async Task<SearchDialogResult> Handle(SearchDialogQuery request, Cancella
             .WhereIf(request.DueAfter.HasValue, x => request.DueAfter <= x.DueAt)
             .WhereIf(request.DueBefore.HasValue, x => x.DueAt <= request.DueBefore)
             .WhereIf(request.Process is not null, x => EF.Functions.ILike(x.Process!, request.Process!))
-            .WhereIf(!request.SystemLabel.IsNullOrEmpty(), x => request.SystemLabel!.Contains(x.EndUserContext.SystemLabelId))
+            .WhereIf(!request.SystemLabel.IsNullOrEmpty(), x =>
+                request.SystemLabel!.Any(label =>
+                    x.EndUserContext.DialogEndUserContextSystemLabels
+                        .Any(sl => sl.SystemLabelId == label)))
```

Comment:
```suggestion
            .WhereIf(!request.SystemLabel.IsNullOrEmpty(), x =>
                request.SystemLabel!.All(label =>
                    x.EndUserContext.DialogEndUserContextSystemLabels
                        .Any(sl => sl.SystemLabelId == label)))
```

Korriger meg om jeg tar feil - men er ikke standarden vår å AND'e query parameterne våre? Så hvis jeg oppgir både bin og notRead, så burde jeg kun få dialoger som har begge disse lablene, ikke kun en av dem?

Reply:
Og hvis jeg oppgir både bin og archive er det PEBCAK 😂